File: capstone/vue/src/services/BeerService.js
import axios from 'axios';

export default {
  getBeers() {
    return axios.get('/beers');
  },
  getBeerById(beerId) {
    return axios.get(`/beers/${beerId}`);
  },
  
  getSavedBeers() {
    return axios.get('/mybeers')
  },

  getRandomBeer() {
    return axios.get('/beers/random')
  },

  getBeersByBrewery(breweryId) {
    return axios.get(`/brewery/beers/${breweryId}`);
  },

  addSavedBeer(beerId) {
    return axios.post(`/mybeers/${beerId}`);
  },

  createBeer(beer) {
    return axios.post('/beers', beer);
  },

  updateBeer(beer) {
    return axios.put(`/beers/${beer.beerId}`, beer);
  },

  deleteSavedBeer(beerId) {
    return axios.delete(`/mybeers/${beerId}`);
  },

  deleteBeer(beerId) {
    return axios.delete(`/beers/${beerId}`);
  }
};

File: capstone/java/src/main/java/com/techelevator/service/BeerService.java
package com.techelevator.service;

import com.techelevator.model.Beer;
import org.springframework.stereotype.Service;

import java.math.BigDecimal;
import java.security.Principal;
import java.util.List;

public interface BeerService {
    List<Beer> getBeers();
    Beer getRandomBeer();
    Beer getBeerById(int beerId);
    List<Beer> getBeersByBreweryId(int breweryId);
    List<Beer> getSavedBeers(Principal principal);
    Beer addBeerToSaved(int beerId, Principal principal);
    Beer createBeer(Beer beer, Principal principal);
    Beer updateBeer(Beer beer, Principal principal);
    void deleteSavedBeer(int beerId, Principal principal);
    void deleteBeer(int beerId, Principal principal);

}
